android/Application/src/main/java/de/tudarmstadt/es/able/PermissionClass.java

```diff
@@ -0,0 +1,86 @@
+package de.tudarmstadt.es.able;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.os.Build;
+import android.provider.Settings;
+import android.support.v4.content.ContextCompat;
+import android.text.TextUtils;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+
+/**
+ * Created by user on 14.02.18.
+ */
+
+public class PermissionClass {
+
+
+
+
+
+    //method to get current activity(ies)
+    public static Activity getActivity() throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, NoSuchFieldException {
+        Class activityThreadClass = Class.forName("android.app.ActivityThread");
+        Object activityThread = null;
+        try {
+            activityThread = activityThreadClass.getMethod("currentActivityThread").invoke(null);
+        } catch (InvocationTargetException e) {
+            e.printStackTrace();
+        }
+        Field activitiesField = activityThreadClass.getDeclaredField("mActivities");
```

Comment:
... even more nasty ;-)

Is this really necessary? Isn't there an easier way?

Reply:
done